src/hooks/useEAS.ts

```diff
@@ -1,23 +1,91 @@
-import { Hex, getAddress } from 'viem';
+import { Address, Hex, getAddress } from 'viem';
 import { useState, useEffect } from 'react';
 import { useToast } from '@chakra-ui/react';
 
-import { useWalletClient } from 'wagmi';
+import { useWalletClient, useSendTransaction, useChainId } from 'wagmi';
+import {
+  Hash,
+  waitForTransaction,
+  sendTransaction,
+  prepareSendTransaction,
+} from '@wagmi/core';
+import { parseEther } from 'viem';
+
 import { useScore } from './index';
 import { getScoreAttestations } from '@/helpers/eas';
+import { ATTESTER_PUBLIC_KEY } from '@/constants';
+import { checkPendingReceipt } from '@/helpers/databaseHelpers';
 
 function useEAS(address: string | Hex | undefined) {
   const toast = useToast();
   const { data: walletClient } = useWalletClient();
+  const chainId = useChainId();
   const { score, meta } = useScore(address);
+  const [ethToUsdPrice, setEthToUsdPrice] = useState(0);
 
   // TODO: FIX TYPES
   const [attestations, setAttestations] = useState(null);
   const [lastAttestation, setLastAttestation] = useState<any>(null);
+  useEffect(() => {
+    async function fetchETHPrice() {
```

Comment:
let's extract this function, we also need it on the backend to check if the user sent the right amount of eth